src/main/java/com/workshop/demo/service/impl/ReviewServiceImpl.java

```diff
@@ -0,0 +1,123 @@
+package com.workshop.demo.service.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import com.workshop.demo.exception.ResourceNotFoundException;
+import com.workshop.demo.exception.BlogapiException;
+import com.workshop.demo.model.Restaurant;
+import com.workshop.demo.model.Review;
+import com.workshop.demo.model.User;
+import com.workshop.demo.payload.ApiResponse;
+import com.workshop.demo.payload.RestaurantResponse;
+import com.workshop.demo.payload.ReviewRequest;
+import com.workshop.demo.repository.RestaurantRepository;
+import com.workshop.demo.repository.ReviewRepository;
+import com.workshop.demo.repository.UserRepository;
+import com.workshop.demo.security.UserPrincipal;
+import com.workshop.demo.service.ReviewService;
+
+public class ReviewServiceImpl implements ReviewService {
+
+    private static final String THIS_REVIEW = " this review";
+
+    private static final String THIS_RESTAURANT = " this restaurant";
+
+    private static final String YOU_DON_T_HAVE_PERMISSION_TO = "You don't have permission to ";
+
+    private static final String ID_STR = "id";
+
+    private static final String REVIEW_STR = "Review";
+
+    private static final String REVIEW_DOES_NOT_BELONG_TO_POSTER = "Review does not belong to poster";
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private ReviewRepository reviewRepository;
+
+    @Autowired
+    private RestaurantRepository restaurantRepository;
+
+    // get all the reviews of the restaurant
+    @Override
+    public Review getAllReviews(String restaurantName) {
```

Comment:
This should return a list of reviews instead of a single Review object.

Reply:
return reviewRepository.findByRestaurantName(restaurantName)
                .orElseThrow(() -> new ResourceNotFoundException(REVIEW_STR, ID_STR, restaurantName));
is this qualified?